main/pipeline/pipeline.cc

```diff
@@ -1060,6 +1062,14 @@ void typecheck(const unique_ptr<core::GlobalState> &gs, vector<ast::ParsedFile>
                WorkerPool &workers, bool cancelable,
                optional<shared_ptr<core::lsp::PreemptionTaskManager>> preemptionManager, bool presorted,
                bool intentionallyLeakASTs) {
+    return typecheck(reinterpret_cast<const unique_ptr<const core::GlobalState> &>(gs), move(what), opts, workers,
+                     cancelable, move(preemptionManager), presorted, intentionallyLeakASTs);
```

Comment:
why not just make `typecheck` take a `const core::GlobalState&` and then delete the overload?

you've already proven that nothing in typecheck needs a mutable state, and we can easily dereference the gs before calling typecheck at the callsites.